client/blocks/image-editor/index.jsx

```diff
@@ -72,14 +88,54 @@ const ImageEditor = React.createClass( {
 		} = this.props;
 
 		if ( newProps.media && ! isEqual( newProps.media, currentMedia ) ) {
+			this.checkForValidAspectRatioSettings();
+
 			this.props.resetAllImageEditorState();
 
 			this.updateFileInfo( newProps.media );
+
+			this.setDefaultAspectRatio();
 		}
 	},
 
 	componentDidMount() {
 		this.updateFileInfo( this.props.media );
+
+		this.setDefaultAspectRatio();
+	},
+
+	checkForValidAspectRatioSettings() {
+		const {
+			allowedAspectRatios,
+			defaultAspectRatio
+		} = this.props;
+
+		if (
+			allowedAspectRatios &&
+			allowedAspectRatios.length !== 1 &&
+			allowedAspectRatios.indexOf( defaultAspectRatio ) === -1
+		) {
+			// If allowedAspectRatios prop is specified and has more than one item,
+			// defaultAspectRatio prop must contain a value which is included in allowedAspectRatios.
+			throw new Error( 'Image Editor props: defaultAspectRatio not found in allowedAspectRatios' );
+		}
+	},
+
+	setDefaultAspectRatio() {
+		const {
+			defaultAspectRatio,
+			allowedAspectRatios
+		} = this.props;
+
+		if ( allowedAspectRatios && allowedAspectRatios.length === 1 ) {
+			this.props.setImageEditorAspectRatio( allowedAspectRatios[ 0 ] );
```

Comment:
My intention here was that if `allowedAspectRatios` contains only single aspect ratio, let it set as a default aspect ratio as well. If somebody puts only a single aspect ratio into `allowedAspectRatios` while passing `defaultAspectRatio` with different value, the value which is in `allowedAspectRatios` will be set as the default one and the function` setDefaultAspectRatio` `return`s after that, so the value from `defaultAspectRatio` won't be even considered.

Do you think it's still a bug?